pkg/controller/nodes/task/handler.go

```diff
@@ -230,8 +243,32 @@ func (t *Handler) Setup(ctx context.Context, sCtx handler.SetupContext) error {
 	return nil
 }
 
-func (t Handler) ResolvePlugin(ctx context.Context, ttype string) (pluginCore.Plugin, error) {
-	p, ok := t.plugins[ttype]
+func (t Handler) ResolvePlugin(ctx context.Context, ttype string, executionConfig v1alpha1.ExecutionConfig) (pluginCore.Plugin, error) {
+	// If the workflow specifies plugin overrides, check to see if any of the specified plugins for that type are
+	// registered in this deployment of flytepropeller.
+	if len(executionConfig.TaskPluginImpls) > 0 && len(executionConfig.TaskPluginImpls[ttype]) > 0 {
```

Comment:
nit: I _think_ the second check is enough but I could be wrong
```suggestion
	if taskPluginImpls := executionConfig.TaskPluginImpls[ttype];len(taskPluginImpls) > 0 {
```